core/src/components/input/test/password-toggle/input.e2e.ts

```diff
@@ -0,0 +1,68 @@
+import { expect } from '@playwright/test';
+import { configs, test } from '@utils/test/playwright';
+
+configs().forEach(({ title, screenshot, config }) => {
+  test.describe(title('input: password toggle button'), () => {
+    test('should render password icon when show-password-toggle', async ({ page }) => {
+      await page.setContent(
+        `
+            <ion-input value="abc" type="password" clear-input="true" show-password-toggle="true"></ion-input>
+        `,
+        config
+      );
+
+      const input = page.locator('ion-input');
+      await expect(input).toHaveScreenshot(screenshot(`input-with-password-toggle`));
+    });
+
+    test('should toggle the input type when pressed', async ({ page }) => {
+      await page.setContent(
+        `
+            <ion-input value="abc" type="password" clear-input="true" show-password-toggle="true"></ion-input>
+        `,
+        config
+      );
+
+      const input = page.locator('ion-input');
+      const nativeInput = page.locator('.native-input');
+      const toggle = input.locator('.input-password-toggle');
+
+      await toggle.click();
+      await page.waitForChanges();
+
+      await expect(nativeInput).toHaveAttribute('type', 'text');
+
+      await toggle.click();
+      await page.waitForChanges();
+
+      await expect(nativeInput).toHaveAttribute('type', 'password');
```

Comment:
This could be a spec test. We don't necessarily benefit from having cross-browser testing here.

Reply:
Makes sense! I will take care of this tomorrow.

Reply:
Done!